taxonomy-editor-frontend/src/pages/editentry/AccumulateAllComponents.jsx

```diff
@@ -29,8 +30,19 @@ const AccumulateAllComponents = ({ id, taxonomyName, branchName }) => {
     errorMessage,
   } = useFetch(url);
   const [nodeObject, setNodeObject] = useState(null); // Storing updates to node
+  const [originalNodeObject, setOriginalNodeObject] = useState(null); // For tracking changes
   const [updateChildren, setUpdateChildren] = useState([]); // Storing updates of children in node
   const [open, setOpen] = useState(false); // Used for Dialog component
+  const [changesMade, setChangesMade] = useState(false); // Used for displaying Fab
+
+  // Tracking changes
+  useEffect(() => {
+    if (nodeObject && originalNodeObject) {
+      const changesMade =
+        JSON.stringify(nodeObject) !== JSON.stringify(originalNodeObject);
```

Comment:
JSON.stringify does not ensure that the strings will have the same shape. Let's use https://www.npmjs.com/package/fast-deep-equal for this. Also, no need for this to be a state, it can be a derived value